source/main.cpp

```diff
@@ -161,6 +155,14 @@ int main()
     // Initialise the micro:bit runtime.
     uBit.init();
 
+    Gigglebot* gigglebot = new Gigglebot(uBit.i2c);
+    Cutebot* cutebot = new Cutebot(uBit.i2c);

```

Comment:
Might want to only construct the second one of the first isn't detected. Then delete the first so there isn't extra memory usage

Reply:
:+1:

Reply:
`warning: deleting object of polymorphic class type 'Gigglebot' which has non-virtual destructor might cause undefined behavior`

YOLO

Reply:
I removed this warning by implementing the virtual destructor